### ISSUE TITLE
[QUESTION] Remaining TODOs for Calibre >= 5?

### ISSUE THREAD

**Issue by pistoletpierre** [apprenticeharper/DeDRM_tools#1363]
Is there a list somewhere of things that still need to be fixed/updated in order for the plugin to work the Calibre 5 and up?

**Comment by 1031982** [comment]
I would think were more likely to get a percentage of completion. I would love to know this.
However, I don't want to potentially create an up-stir of people complaining that's it's not far enough along, or getting mad, etc. I don't know the community here, but lately these days, a lot of people don't seem to understand this stuff takes time and any estimate is just that. On some forums, I have seen a number of developers give estimates only to miss them and have the community attack them.
In the end, I hope this gets updated to Python3 sooner rather then later, but we will get it whenever it's done.

**Comment by pistoletpierre** [comment]
Makes sense. I'm interested in contributing to the effort. If the developers aren't keen on making a full list publicly available, perhaps I could pop tasks from a list that they manage privately?

**Comment by null-cipher** [comment]
I'd also be interested to get some pointers for this. I have a bit of experience converting python2 codebases to python3, but I'm not sure where to start with this repo as it's not super clear which files have been fully converted yet or not.

**Comment by apprenticeharper** [comment]
It's not a simple conversion. Each file mus be tested and updated, as there's lots of stuff that still needs to be in byte strings.

Pick a file. Test it. If you find a problem, fix it and submit a pull request. Do make sure you test your fix on Mac and Windows. Test it stand-alone, and as a plugin.

When I can, I'm working through them alphabetically. :)

Conversion of stand-alone to use pycryptodome instead of pycrypto would also be appreciate, so long as you don't break the plugin.

**Comment by 1031982** [comment]
I counted over 50 python files, now I wish I knew how to program with python and had a Mac to help out.
However long it takes, thank you for the hard work to allow people to use content they buy as they wish (legally of course).
